scripts/install_ebpf.psm1

```diff
@@ -219,48 +146,171 @@ function Install-eBPFComponents
+    # Test is the VC debug runtime DLLs are present in the working directory (indicating a debug build).
+    $VCDebugRuntime = $VCDebugRuntime | Where-Object { Test-Path (Join-Path $WorkingDirectory $_) }
+    if (-not $VCDebugRuntime) {
+        Write-Log("VC debug runtime DLLs not found in the working directory (i.e. release build). Skipping this step.") -ForegroundColor Yellow
+    } else {
+        $system32Path = Join-Path $env:SystemRoot "System32"
+        $VCDebugRuntime | ForEach-Object {
+            $sourcePath = Join-Path $WorkingDirectory $_
+            $destinationPath = Join-Path $system32Path $_
+            Write-Log("Copying '$sourcePath' to '$destinationPath'...")
+            Copy-Item -Path $sourcePath -Destination $destinationPath -Force
+        }
+        Write-Log("VC debug runtime DLLs copied successfully!") -ForegroundColor Green
+    }
+
+    # Install the MSI package.
+    $arguments = "/i $MsiPath ADDLOCAL=ALL /qn /norestart /l*v msi-install.log"
+    Write-Log("Installing the eBPF MSI package: 'msiexec.exe $arguments'...")
+    $process = Start-Process -FilePath msiexec.exe -ArgumentList $arguments -Wait -PassThru
+    if ($process.ExitCode -ne 0) {
+        Write-Log("MSI installation FAILED. Exit code: $($process.ExitCode)") -ForegroundColor Red
+
+        # For clear readability within the CICD pipeline and final uploaded log output,
+        # read each line of the log file and print it (otherwise all the log content is printed as a single line).
+        Write-Log("Contents of msi-install.log:")
+        Get-Content -Path "msi-install.log" | ForEach-Object {
+            Write-Log($_)
+        }
+        throw ("MSI installation FAILED. Exit code: $($process.ExitCode)")
     }
-    if (Test-Path -Path "testing\testing") {
-        Copy-Item testing\testing\*.sys -Destination "$Env:systemroot\system32\drivers" -Force -ErrorAction Stop 2>&1 | Write-Log
+    Write-Log("eBPF MSI installation completed successfully!") -ForegroundColor Green
+
+    # Install the extra drivers that are not installed by the MSI package.
+    $EbpfDrivers.GetEnumerator() | ForEach-Object {
+        if (-not $_.Value.InstalledByMsi) {
+            $driverPath = if (Test-Path -Path ("$pwd\{0}" -f $_.Value.Name)) {
+                "$pwd\{0}" -f $_.Value.Name
+            } elseif (Test-Path -Path ("$pwd\drivers\{0}" -f $_.Value.Name)) {
+                "$pwd\drivers\{0}" -f $_.Value.Name
+            } else {
+                throw ("Driver file not found for $($_.Key).")
+            }
+            Write-Log("Installing $($_.Key)...") -ForegroundColor Green
+            sc.exe create $_.Key type=kernel start=demand binpath=$driverPath 2>&1 | Write-Log
+            if ($LASTEXITCODE -ne 0) {
+                throw ("Failed to create $($_.Key) driver.")
+            } else {
+                Write-Log("$($_.Key) driver created.") -ForegroundColor Green
+                # Start the service
```

Comment:
```suggestion
                # Start the service.
```
nit: I'm noticing that some comments in in punctuation and others like this one don't.  Would be good to be consistent throughout this file.